01/ht/edithpifpaf/01/01.md

```diff
@@ -0,0 +1,31 @@
+function calc(x) {
+  return function calculate(z) {
+    if (z==='+') {
+      return function sum(y) {
+      return isNumber(x) + isNumber(y);
+      }
+    } else if (z==='-') {
+      return function subtract(y) {
+        return isNumber(x) - isNumber(y);
+      }
+    } else if (z==='*') {
+      return function multiply(y) {
+        return isNumber(x) * isNumber(y);
+      }
+    } else if (z==='/') {
+      return function divide(y) {
+        return isNumber(x) / isNumber(y);
+      }
+   } else {
+      alert (' Not an operator! ');
+    }
+  }
+}
+
+function isNumber(number) {
```

Comment:
why is it called `isNumber`. Naming convention says that if function starts with `is` - it's checker which should return boolean value